### PR TITLE
SAK-41853 - Add support for LTI 1.1 Launch Type

### PR DESCRIPTION
**Pull request by csev**

### CODE REVIEW

File: basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java [hunk 1]
```diff
@@ -990,6 +995,14 @@ public static void addGlobalData(Site site, Properties props, Properties custom,
 		setProperty(toolProps, "state", state);  // So far LTI 1.3 only
 		setProperty(toolProps, "nonce", nonce);  // So far LTI 1.3 only
 
+		// LTI 1.1.2
+		String tool_state = (String) tool.get("tool_state");
+		if ( ! StringUtils.isEmpty(tool_state) ) setProperty(ltiProps, "tool_state", tool_state);
```
Comment (ottenhoff) [inline basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java]: isNotEmpty is better than ! isEmpty

File: basiclti/basiclti-impl/src/java/org/sakaiproject/basiclti/impl/BasicLTISecurityServiceImpl.java [hunk 2]
```diff
@@ -297,6 +297,40 @@ private void redirectOIDC(HttpServletRequest req, HttpServletResponse res,
...
+		if ( lti11_launch_type.equals(LTIService.LTI11_LAUNCH_TYPE_LTI112) ||
+					SakaiBLTIUtil.BASICLTI_LTI11_LAUNCH_TYPE_LTI112.equals(default_launch_type) ) {
+
+			String tool_state = req.getParameter("tool_state");
+			if ( StringUtils.isEmpty(tool_state) ) {
+				// req.getRequestURL()=http://localhost:8080/access/basiclti/site/85fd092b-1755-4aa9-8abc-e6549527dce0/content:0
+				// req.getRequestURI()=/access/basiclti/site/85fd092b-1755-4aa9-8abc-e6549527dce0/content:0
+				String platform_state = req.getRequestURI();
+				String query_string = req.getQueryString();
+					if ( StringUtils.isNotEmpty(query_string)) {
```
Comment (ottenhoff) [inline basiclti/basiclti-impl/src/java/org/sakaiproject/basiclti/impl/BasicLTISecurityServiceImpl.java]: tabs/spacing looks off in this section

Comment (csev) [reply basiclti/basiclti-impl/src/java/org/sakaiproject/basiclti/impl/BasicLTISecurityServiceImpl.java]: Thanks - Took a bit to figure out it was just one line :)  I looking for spaces.

File: basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java (hunk 1 above)
Comment (csev) [reply basiclti/basiclti-common/src/java/org/sakaiproject/basiclti/util/SakaiBLTIUtil.java]: Thanks.

**Pull request merged**